WalletWasabi/WabiSabi/Backend/CoinJoinMempoolManager.cs

```diff
@@ -0,0 +1,60 @@
+using NBitcoin;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Bases;
+using WalletWasabi.WabiSabi.Backend.Rounds.CoinJoinStorage;
+using System.Collections.Immutable;
+using WalletWasabi.BitcoinCore.Mempool;
+
+namespace WalletWasabi.WabiSabi.Backend;
+
+public class CoinJoinMempoolManager : IDisposable
+{
+	private bool _disposedValue;
+
+	public CoinJoinMempoolManager(ICoinJoinIdStore coinJoinIdStore)
+	{
+		CoinJoinIdStore = coinJoinIdStore;
+	}
+
+	private ICoinJoinIdStore CoinJoinIdStore { get; }
+	private MempoolMirror? Mempool { get; set; }
+	public ImmutableArray<uint256> CoinJoinIds { get; private set; } = ImmutableArray.Create<uint256>();
+
+	public void RegisterMempoolProvider(MempoolMirror mempool)
```

Comment:
This is an anti-pattern that we should never add because it creates lots of problems and then it is Kiminuo who invest a lot of time to remove. 

When a class  requires a dependency without which it cannot do anything, that dependency has to be provided in the constructor and not in a setter.

Reply:
I tried that. Not possible. MempoolMirror is only created where this method is called. On the other hand CoinJoinMempoolManager required earlier for injections. I don't know a better solution.

Reply:
@kiminuo do you have any idea to make this better without significant refactoring of other components?

Reply:
Zes I took your approach, I think it is much better now.